### PR TITLE
add e2e multihoming tests for localnet network

### PR DESCRIPTION
**Pull request by cathy-zhou**
**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by cathy-zhou** [comment]
@maiqueb note that I am not able to add any connectivity testing cases as connectivity on localnet network is through underlay.

**Comment by maiqueb** [comment]
> @maiqueb note that I am not able to add any connectivity testing cases as connectivity on localnet network is through underlay.

hm, you could force both pods to be scheduled in the same node, right ? @cathy-zhou

**Comment by cathy-zhou** [comment]
> hm, you could force both pods to be scheduled in the same node, right ? 
don't understand how that helps. OVN flows would still direct the traffic to underlay.

File: test/e2e/multihoming.go [hunk 1]
```diff
@@ -24,11 +24,13 @@ import (
...
-		secondaryNetworkName       = "tenant-blue"
-		secondaryFlatL2IgnoreCIDR  = "10.128.0.0/29"
-		secondaryFlatL2NetworkCIDR = "10.128.0.0/24"
+		podName                      = "tinypod"
+		secondaryNetworkCIDR         = "10.128.0.0/16"
+		secondaryNetworkName         = "tenant-blue"
+		secondaryFlatL2IgnoreCIDR    = "10.128.0.0/29"
+		secondaryFlatL2NetworkCIDR   = "10.128.0.0/24"
+		secondaryLocalnetIgnoreCIDR  = "60.128.0.0/29"
+		secondaryLocalnetNetworkCIDR = "60.128.0.0/24"
```
Comment (maiqueb) [inline test/e2e/multihoming.go:33]: do we really need a different CIDR for this type of networks ?

File: test/e2e/multihoming.go [hunk 2]
```diff
@@ -507,6 +614,14 @@ func namespacedName(namespace, name string) string {
...
 
 func inRange(cidr string, ip string) error {
+	// cidr is in the form of either <subnet>/<prefix> or <subnet>/<prefix>/<nodePrefix>
+	s := strings.Split(cidr, "/")
+	if len(s) > 3 || len(s) < 2 {
+		return fmt.Errorf("invalid cidr %s", cidr)
+	} else if len(s) == 3 {
+		// this is cidr of layer3 network
+		cidr = s[0] + "/" + s[1] 
+	}
```
Comment (maiqueb) [inline test/e2e/multihoming.go:624]: I prefer to do the opposite: have the caller always pass a valid CIDR to this function. 

I'm doing something like this in https://github.com/ovn-org/ovn-kubernetes/pull/3442/files#diff-8a0101c1714fa2f18fcb47cc2f71b4a820bc6e88fb23f0fcb557abafbfb3b924R201 

Let's see what the maintainers think.

**Review by maiqueb — COMMENTED** [review]
I just have a couple of tiny comments, other than that, looks really good.

Thank you.

File: test/e2e/multihoming.go (hunk 2 above)
Comment (jcaamano) [reply test/e2e/multihoming.go:624]: Yup, certainly better to have a different specific function to parse the CIDR from `<subnet>/<prefix>/<nodePrefix>` to be used independently from this `InRange` function.

File: test/e2e/multihoming.go (hunk 1 above)
Comment (cathy-zhou) [reply test/e2e/multihoming.go:33]: we don't. but I thought we need to allow it. If one day we are able to somehow make use of underlay networking, we may need specific CIDR/VLANID combination.

Comment (maiqueb) [reply test/e2e/multihoming.go:33]: IMO, we should add these once that day comes then.

I won't insist further.

File: test/e2e/multihoming.go (hunk 2 above)
Comment (cathy-zhou) [reply test/e2e/multihoming.go]: fixed

File: test/e2e/multihoming.go [hunk 3]
```diff
@@ -244,7 +312,9 @@ var _ = Describe("Multi Homing", func() {
 						By("asserting the server pod has an IP from the configured range")
 						Expect(netStatus[0].IPs).NotTo(BeEmpty())
 						serverIP = netStatus[0].IPs[i]
-						Expect(inRange(popCIDRsPerNodePrefix(subnet, netPrefixLengthPerNode), serverIP)).To(Succeed())
+						cidrRange, err := getCIDRRange(subnet)
```
Comment (maiqueb) [inline test/e2e/multihoming.go:315]: why are you making this change  ? It works without it, doesn't it ?

Comment (cathy-zhou) [reply test/e2e/multihoming.go:315]: My thought is that get cidrRange based on form of cidr is better than assuming that the nodeSubnet is 24. no strong opinion though, I can change it back if you prefer.

Comment (maiqueb) [reply test/e2e/multihoming.go:315]: nothing is being assumed, we're using a constant whose name documents what is happening. 

You can even update the code to pass the attribute via networkConfig in each test if you prefer (and only set it for layer3).

IMO, that's preferable.

Comment (cathy-zhou) [reply test/e2e/multihoming.go]: ok, reverted.

**Review by maiqueb — APPROVED** [review]
Thank you.

Question: would it make sense to add a traffic test without a vlanID defined ?

**Pull request merged**